src/hoverifier.ts

```diff
@@ -208,23 +211,17 @@ export interface HoverState {
 
 /**
  * @template C Extra context for the hovered token.
+ * @template D The type of the hover content data.
+ * @template A The type of an action.
  */
-interface InternalHoverifierState<C extends object> {
-    hoverOrError?: typeof LOADING | HoverMerged | null | ErrorLike
-    definitionURLOrError?: typeof LOADING | { jumpURL: string } | null | ErrorLike
-    referencesURL?: string | null
+interface InternalHoverifierState<C extends object, D, A> {
+    hoverOrError?: typeof LOADING | (HoverAttachment & D) | null | ErrorLike
 
     hoverOverlayIsFixed: boolean
 
     /** The desired position of the hover overlay */
     hoverOverlayPosition?: { left: number; top: number }
 
-    /**
-     * Whether the user has clicked the go to definition button for the current overlay yet,
-     * and whether he pressed Ctrl/Cmd while doing it to open it in a new tab or not.
-     */
-    clickedGoToDefinition: false | 'same-tab' | 'new-tab'
```

Comment:
If this is removed, how can we abstract showing the loader on the button if the button is clicked before the definition loaded and jumping to it once it's there?

Reply:
It’s implemented and tested in Sourcegraph in the same way we do it for other ActionItems. It’s in the linked PR.

Reply:
It’s in shared/src/hover/actions.ts. We can go over it on our call today.

I’m optimizing for the immediate use case of Sourcegraph needing to use this, not for other consumers that don’t exist yet. But I still tried to keep this library a useful abstraction with more tests than before.